src/comp/pages/UserDetailsPage.tsx

```diff
@@ -99,52 +93,60 @@ class UserDetailsPage extends React.Component<
     const empty = { userDetails: <></>, devExperience: <></> };
     let phoneElement: Element;
     let pcElement: Element;
-    if (this.state.isPhone) {
-      phoneElement = insertContent;
-      pcElement = empty;
-    } else {
-      pcElement = insertContent;
-      phoneElement = empty;
-    }
+    const createElements = (isPhone: boolean) => {
+      if (isPhone) {
+        phoneElement = insertContent;
+        pcElement = empty;
+      } else {
+        pcElement = insertContent;
+        phoneElement = empty;
+      }
+    };
     return (
-      <>
-        <EventListener target="window" onResize={this.handleResize} />
-        <div className={styles.profilePageWrapper}>
-          <div className={styles.profileWrapper}>
-            <div className={styles.profileImageWrapper}>
-              <img
-                src={this.state.profileImageUrl}
-                alt={this.state.name}
-                className={styles.profileImage}
-              />
-            </div>
-            <div className={styles.followButton}>
-              <FollowButton
-                isFollowing={this.checkFollowing()}
-                updateFollowingList={this.updateList}
-              />
-            </div>
-            {pcElement.userDetails}
-            <div className={styles.accountWrapper}>
-              <LinkedAccounts
-                accounts={this.state.accounts}
-                linkedAccountsList={this.state.linkedAccountsList}
-                isPhone={this.state.isPhone}
-              />
-            </div>
-          </div>
-          <div className={styles.profileDetailsWrapper}>
-            <h1>{this.state.name}</h1>
-            <Divider />
-            {phoneElement.userDetails}
-            <div className={styles.bio}>
-              <p>{this.state.bio}</p>
-            </div>
-            {pcElement.devExperience}
-          </div>
-        </div>
-        {phoneElement.devExperience}
-      </>
+      <PhoneContext.Consumer>
```

Comment:
こっちも `useContext` 使わないんですか

Reply:
classコンポーネントなので使えないです。
```
Error: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://fb.me/react-invalid-hook-call for tips about how to debug and fix this problem.```

Reply:
わかった Reolve する